docs/documentation/dependent-resources.md

```diff
@@ -38,187 +36,203 @@ match -- No --> Update --> Done
-example if calling an API needs to happen if a service is already up and running 
-(think configuring a running DB instance). 
+mode is available. In practice this means that the developer is responsible to initializing and managing and
+calling `reconcile` method. However, this gives possibility for developers to fully customize the workflow for
+reconciliation. Like setting conditions (if creation of a resource is desired only in certain situations). Also, for
+example if calling an API needs to happen if a service is already up and running
+(think configuring a running DB instance).
 
 The following sample is equivalent to the one above with managed dependent resources:
 
 ```java
+
 @ControllerConfiguration
 public class WebPageStandaloneDependentsReconciler
-    implements Reconciler<WebPage>, ErrorStatusHandler<WebPage>, EventSourceInitializer<WebPage> {
-    
-  private KubernetesDependentResource<ConfigMap, WebPage> configMapDR;
-  private KubernetesDependentResource<Deployment, WebPage> deploymentDR;
-  private KubernetesDependentResource<Service, WebPage> serviceDR;
-
-  public WebPageStandaloneDependentsReconciler(KubernetesClient kubernetesClient) {
-    // 1.
-    createDependentResources(kubernetesClient);
-  }
-
-  @Override
-  public List<EventSource> prepareEventSources(EventSourceContext<WebPage> context) {
-    // 2.  
-    return List.of(
-      configMapDR.initEventSource(context),
-      deploymentDR.initEventSource(context),
-      serviceDR.initEventSource(context));
-  }
-
-  @Override
-  public UpdateControl<WebPage> reconcile(WebPage webPage, Context<WebPage> context)
-      throws Exception {
-
-    // 3.  
-    configMapDR.reconcile(webPage, context);
-    deploymentDR.reconcile(webPage, context);
-    serviceDR.reconcile(webPage, context);
-
-    // 4.
-    webPage.setStatus(
-       createStatus(configMapDR.getResource(webPage).orElseThrow().getMetadata().getName()));
-    return UpdateControl.updateStatus(webPage);
-  }
-
-  private void createDependentResources(KubernetesClient client) {
-    this.configMapDR = new ConfigMapDependentResource();
-    this.configMapDR.setKubernetesClient(client);
-    configMapDR.configureWith(new KubernetesDependentResourceConfig()
-      .setLabelSelector(DEPENDENT_RESOURCE_LABEL_SELECTOR));
-
-    this.deploymentDR = new DeploymentDependentResource();
-    deploymentDR.setKubernetesClient(client);
-    deploymentDR.configureWith(new KubernetesDependentResourceConfig()
-      .setLabelSelector(DEPENDENT_RESOURCE_LABEL_SELECTOR));
-
-    this.serviceDR = new ServiceDependentResource();
-    serviceDR.setKubernetesClient(client);
-    serviceDR.configureWith(new KubernetesDependentResourceConfig()
-      .setLabelSelector(DEPENDENT_RESOURCE_LABEL_SELECTOR));
-  }
-  
-  // omitted code
+        implements Reconciler<WebPage>, ErrorStatusHandler<WebPage>, EventSourceInitializer<WebPage> {
```

Comment:
Isn't there a way to include code directly from the repo instead of having to copy it here?

Reply:
Good, point will check!

Reply:
https://emgithub.com/ 
can do the trick, the problem is that it is not very well formatted (while it is on that page directly), it is not in the resulting jekyll page.
Also I would use permalinks, since any reformat / change can mess this up if we want to use it. Since we reference the code snippet based on line numbers.

Reply:
![image](https://user-images.githubusercontent.com/5443269/160084404-796d9ec0-c764-4ba3-aff7-846441be96d9.png)

Reply:
So it messes up the formatting, I was not able to fix it for now. What do you think should we use this? @metacosm

Reply:
Also we cannot ommit code from middle.

Reply:
Maybe just pointing to the code is enough?

Reply:
That I already added, see "See the full source code sample here" parts.

Reply:
Ahh you mean only that, I think that is very hard to read.

Reply:
I'd prefer if we could embed the code but if we can't do it nicely, then copying the code in the doc is more problematic because then it needs to be maintained and risk being outdated, causing user confusion.

Reply:
I would also prefer that, it's just not formatted now :(
And was not able to fix it for now.

I would merge this as it is, to be consistent with other parts of the docs, and we can migrate / take a look on the embedding & formatting isses seratelly, pretty sure that can be fixed somehow.

Reply:
And will migrate all the docs to that in a separate PR.